flamingo/scripts/bh_luminosities.py

```diff
@@ -0,0 +1,115 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+from swiftsimio import load
+
+from unyt import mh, cm, Gyr
+from matplotlib.colors import LogNorm
+from matplotlib.animation import FuncAnimation
+
+# Set the limits of the figure.
+mass_bounds = [1e5,3e10] 
+value_bounds = [1e38,1e48]
+bins=25
+def_value = -1.
+
+def get_data(filename):
+
+    data = load(filename)
+
+    masses = data.black_holes.subgrid_masses.to("Msun")
+    rad_effs = np.array(data.black_holes.radiative_efficiencies.value)
+    values = 6.44*10**23*9*10**16*10**4*np.array(data.black_holes.accretion_rates.value)*rad_effs
```

Comment:
Please leave some comments explaining all the various factors that appear here. It would also help to split this up into multiple lines.

Reply:
I am now much cleaner with units throughout. Hence in this particular example comments may not be necessary, as it should be fairly clear from the lines what is going on:
`   rad_effs = data.black_holes.radiative_efficiencies
    accr_rates = data.black_holes.accretion_rates.convert_to_units(unyt.kg / unyt.s)
    values = rad_effs * speed_of_light ** 2 * accr_rates
    values = values.convert_to_units(unyt.erg / unyt.s)`
where `speed_of_light` is introduced earlier as `from unyt import speed_of_light`. Having fixed the units, I now have a couple of issues. In this cript (as well as `bh_jet_powers.py`), I get the following error, which I don't understand:
`  File "/cosma/home/durham/dc-husk1/pipeline-configs-spin-jets/flamingo/scripts/bh_radio_luminosity_functions.py", line 29, in get_data
    jet_powers = jet_effs * speed_of_light ** 2 * accr_rates
  File "/cosma/local/Python/3.6.5/lib/python3.6/site-packages/unyt/array.py", line 1674, in __array_ufunc__
    ret_class = _get_binary_op_return_class(type(i0), type(i1))
  File "/cosma/local/Python/3.6.5/lib/python3.6/site-packages/unyt/array.py", line 2265, in _get_binary_op_return_class
    "Received operand types (%s) and (%s)" % (cls1, cls2)
RuntimeError: Undefined operation for a unyt_array subclass. Received operand types (<class 'swiftsimio.objects.cosmo_array'>) and (<class 'NoneType'>)`

In `bh_merger_mass_fractions.py`, I calculate the merger mass fractions as 
`   masses = data.black_holes.subgrid_masses.to("Msun")
    accreted_masses = data.black_holes.total_accreted_masses.to("Msun")
    values = 1.0 - accreted_masses / masses`
This doesn't work because the field `data.black_holes.total_accreted_masses` appears to have units of `"Msun/Gyr"`, i.e. units of accretion rate, and I am confused as to why.

Reply:
```
accr_rates = data.black_holes.accretion_rates.convert_to_units(unyt.kg / unyt.s)
```
will lead to `accr_rates == None`. `convert_to_units()` converts an array in-place and does not have a return value. You want to use `to(unyt.kg/unyt.s)` instead, which does return a copy of the array in those units. That will solve your first issue.

For the second one: it looks like the dataset is indeed stored with those weird units in the snapshot file. This is probably a bug on the SWIFT side. Thanks for pointing this out!

Reply:
Thanks, `to(unyt.kg/unyt.s)` does indeed work.